libunwind/src/UnwindCursor.hpp

```diff
@@ -2817,7 +2817,7 @@ bool UnwindCursor<A, R>::setInfoForSigReturn(Registers_s390x &) {
   const pint_t pc = static_cast<pint_t>(this->getReg(UNW_REG_IP));
   // The PC might contain an invalid address if the unwind info is bad, so
   // directly accessing it could cause a SIGSEGV.
-  if (!isReadableAddr(pc) || !isReadableAddr(pc + 2))
+  if (!isReadableAddr(pc))
```

Comment:
We might get a false-negative `return false` here if the instruction happens to be the last four bytes before the end of the page boundary but I don't think it it matters.

Reply:
I think that's probably fine, as long as we aren't hitting a SIGSEGV.